src/Microsoft.DotNet.Darc/src/Darc/Operations/GetDependencyGraphOperation.cs

```diff
@@ -193,20 +277,60 @@ private string GetGraphVizNodeName(DependencyGraphNode node)
             return GetSimpleRepoName(node.RepoUri).Replace("-", "") + node.Commit;
         }
 
-        private void LogGraphViz(DependencyGraph graph)
+        private async void LogGraphViz(DependencyGraph graph)
         {
-            Console.WriteLine("digraph repositoryGraph {");
-            Console.WriteLine("    node [shape=record]");
-            foreach (DependencyGraphNode node in graph.Nodes)
+            string directory = Path.GetDirectoryName(_options.GraphVizOutputFile);
+            if (!Directory.Exists(directory))
             {
-                Console.WriteLine($"    {GetGraphVizNodeName(node)}[label=\"{GetSimpleRepoName(node.RepoUri)}\\n{node.Commit.Substring(0, 5)}\"];");
-                foreach (DependencyGraphNode childNode in node.Children)
+                Directory.CreateDirectory(directory);
+            }
+
+            using (StreamWriter writer = new StreamWriter(_options.GraphVizOutputFile))
+            {
+                await writer.WriteLineAsync("digraph repositoryGraph {");
+                await writer.WriteLineAsync("    node [shape=record]");
+                foreach (DependencyGraphNode node in graph.Nodes)
                 {
-                    Console.WriteLine($"    {GetGraphVizNodeName(node)} -> {GetGraphVizNodeName(childNode)}");
+                    string buildString = "";
+                    if (node.ContributingBuilds != null && node.ContributingBuilds.Any())
+                    {
+                        Build newestBuild = node.ContributingBuilds.OrderByDescending(b => b.DateProduced).First();
+                        buildString = $"\\n{newestBuild.DateProduced.Value.ToLocalTime().ToString("g")}";
+                    }
+
+                    string diffString = "";
+                    GitDiff diffFrom = node.DiffFrom;
+                    if (diffFrom != null)
+                    {
+                        if (!diffFrom.Valid)
+                        {
+                            diffString = "\\ndiff unknown";
+                        }
+                        else if (diffFrom.Ahead != 0 || diffFrom.Behind != 0)
+                        {
+                            if (node.DiffFrom.Ahead != 0)
+                            {
+                                diffString += $"\\nahead: {node.DiffFrom.Ahead} commits";
+                            }
+                            if (node.DiffFrom.Behind != 0)
+                            {
+                                diffString += $"\\nbehind: {node.DiffFrom.Behind} commits";
+                            }
+                        }
+                        else
+                        {
+                            diffString = "\\nlatest";
+                        }
+                    }
+                    await writer.WriteLineAsync($"    {GetGraphVizNodeName(node)}[label=\"{GetSimpleRepoName(node.RepoUri)}\\n{node.Commit.Substring(0, 5)}{buildString}{diffString}\"];");
```

Comment:
These lines are so long I can't really read them in git. :-) Can we break it down somehow? Maybe using verbatim strings with literal newlines might help.